src/server/bitops_family.cc

```diff
@@ -577,12 +593,488 @@ void BitCount(CmdArgList args, ConnectionContext* cntx) {
+  bool ShouldCommit() const {
+    return should_commit_;
+  }
+
+ private:
+  Overflow overflow_;
+  std::string value_;
+  bool should_commit_ = false;
+};
+
+using CommandList = std::vector<Command>;
+
+class StateExecutor {
+ public:
+  StateExecutor(ElementAccess access, EngineShard* shard) : access_{access}, shard_(shard) {
+  }
+
+  OpResult<std::vector<ResultType>> Execute(CommandList& commands) {
+    auto res = access_.Exists(shard_);
+    if (!res) {
+      return {OpStatus::WRONG_TYPE};
+    }
+    std::string value;
+    if (*res) {
+      access_.Find(shard_);
+      value = access_.Value();
+    }
+
+    std::vector<ResultType> results;
+    CommandApplyVisitor visitor(std::move(value));
+    for (auto& command : commands) {
+      auto res = std::visit(visitor, command);
+      if (res) {
+        results.push_back(*res);
+      }
+    }
+
+    if (visitor.ShouldCommit()) {
+      access_.Find(shard_);
+      access_.Commit(visitor.Value());
+    }
+
+    return results;
+  }
+
+ private:
+  ElementAccess access_;
+  EngineShard* shard_;
+};
+
+nonstd::expected<CommonAttributes, std::string> ParseCommonAttr(CmdArgParser& parser) {
+  CommonAttributes parsed;
+  using nonstd::make_unexpected;
+  if (!parser.HasAtLeast(2)) {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  auto encoding = parser.ToUpper().Next();
+  if (absl::StartsWith(encoding, "U")) {
+    parsed.type = EncodingType::UINT;
+  } else if (absl::StartsWith(encoding, "I")) {
+    parsed.type = EncodingType::INT;
+  } else {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  std::string_view bits = encoding;
+  bits = bits.substr(1);
+
+  if (!absl::SimpleAtoi(bits, &parsed.encoding_bit_size)) {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  if (parsed.encoding_bit_size <= 0 || parsed.encoding_bit_size > 64) {
+    return make_unexpected(
+        "invalid bitfield type. use something like i16 u8. note that u64 is not supported but i64 "
+        "is.");
+  }
+
+  if (parsed.encoding_bit_size == 64 && parsed.type == EncodingType::UINT) {
+    return make_unexpected(
+        "invalid bitfield type. use something like i16 u8. note that u64 is not supported but i64 "
+        "is.");
+  }
+
+  auto offset_proxy = parser.Next();
+  std::string_view offset_str = offset_proxy;
+  bool is_proxy = false;
+  if (absl::StartsWith(offset_str, "#")) {
+    offset_str = offset_str.substr(1);
+    is_proxy = true;
+  }
+  if (!absl::SimpleAtoi(offset_str, &parsed.offset)) {
+    return make_unexpected(kSyntaxErr);
+  }
+  if (is_proxy) {
+    parsed.offset = parsed.offset * parsed.encoding_bit_size;
+  }
+  return parsed;
+}
+
+nonstd::expected<int64_t, std::string> ParseValue(CmdArgParser& parser) {
+  using nonstd::make_unexpected;
+  if (!parser.HasAtLeast(1)) {
+    return make_unexpected(kSyntaxErr);
+  }
+
+  auto value = parser.ToUpper().Next();
+  int64_t res{0};
+  if (!absl::SimpleAtoi(value, &res)) {
+    return make_unexpected(kSyntaxErr);
+  }
+  return res;
+}
```

Comment:
This is just `parser.Next<int64_t>()`... Please see my previous comment about silent failures, because without it might be inconvenient to actually use the parser

Reply:
Yes, this works here nicely, in fact, `ParseValue` can be eliminated all together(which I did).